<launch>
    <!-- Launch the gps_to_odom node -->
    <node name="gps_to_odom" pkg="first_project" type="gps_to_odom" output="screen">
        <param name="lat_ref" value="48.8582" />
        <param name="lon_ref" value="2.2945" />
        <param name="alt_ref" value="100.0" />
        <remap from="/gps/fix" to="/fix" />
           <!-- This remap is for GPS odometry data -->
        <remap from="/gps_odom" to="/gps_odom_to_tf/input_odom"/>
        <remap from="/gps_odom" to="/wheel_odom_to_tf/input_odom"/>
    </node>

    <!-- Launch node for wheel odometry -->
    <node name="wheel_odom_to_tf" pkg="first_project" type="odom_to_tf" output="screen">
        <param name="root_frame" value="world"/>
        <param name="child_frame" value="wheel_odom"/>
    </node>

    <!-- Launch node for GPS odometry -->
    <node name="gps_odom_to_tf" pkg="first_project" type="odom_to_tf" output="screen">
        <param name="root_frame" value="world"/>
        <param name="child_frame" value="gps_odom"/>
    </node>
</launch>
